### PR TITLE
CS: crypto-bad-mac: populate result message, location, and stack.

### PR DESCRIPTION
**Pull request by lgolding**
Finish implementing the conversion of Contrast Security's `crypto-bad-mac` finding to a SARIF result.

### CODE REVIEW

**Comment by lgolding** [comment]
@michaelcfanning is added to the review. #Closed

**Comment by lgolding** [comment]
@harkohli is added to the review. #Closed

**Comment by lgolding** [comment]
@EasyRhinoMSFT is added to the review. #Closed

**Comment by lgolding** [comment]
@harkohli is added to the review. #Closed

**Comment by lgolding** [comment]
@harleenkohli is added to the review. #Closed

File: src/Sarif.Converters/ContrastSecurityConverter.cs [hunk 1]
```diff
@@ -324,7 +325,46 @@ private Result ConstructAuthorizationRulesMissingDenyResult(ContrastLogReader.Co
...
+                {
+                    resultLocation.LogicalLocation.FullyQualifiedName,
+                    insecureClassName
+                }
+            };
+
+            return result;
+        }
+
+        private static readonly Regex ClassNameFromCtorRegex = new Regex(@"\.([^.]+)\.\.ctor\(\)$");
```
Comment (michaelcfanning) [inline src/Sarif.Converters/ContrastSecurityConverter.cs]: >ClassNameFromCtorRegex [](start = 38, length = 22)

you should make this a compiled regex with explicit capture for best performance.

Comment (lgolding) [reply src/Sarif.Converters/ContrastSecurityConverter.cs]: Fixed.

---
In reply to: [281669231](https://github.com/Microsoft/sarif-sdk/pull/1477#discussion_r281669231) [](ancestors = 281669231)

**Pull request merged**